cpp/arcticdb/version/version_core.cpp

```diff
@@ -858,14 +858,18 @@ void check_incompletes_index_ranges_dont_overlap(const std::shared_ptr<PipelineC
                     "Cannot append staged segments to existing data as incomplete segment contains index value < existing data: {} <= {}",
                     it->slice_and_key().key().start_time(),
                     *last_existing_index_value);
-            unique_timestamp_ranges.insert({it->slice_and_key().key().start_time(), it->slice_and_key().key().end_time()});
-        }
+            auto [_, inserted] = unique_timestamp_ranges.insert({it->slice_and_key().key().start_time(), it->slice_and_key().key().end_time()});
+            // This is correct because incomplete segments aren't column sliced
+            sorting::check<ErrorCode::E_UNSORTED_DATA>(
+                    inserted,
+                    "Cannot finalize staged data as incomplete segments overlap one another");
```

Comment:
Should be possible to make the error message describe the overlap?

Reply:
Added, and also made all index timestamps human readable